libraries/standard/mqtt/doc/doxygen/pages.txt

```diff
@@ -0,0 +1,260 @@
+@snippet mqtt_state.h declare_mqtt_publishtoresend
+@copydoc MQTT_PublishToResend
+
+@page mqtt_getconnectpacketsize_function MQTT_GetConnectPacketSize
+@snippet mqtt_lightweight.h declare_mqtt_getconnectpacketsize
+@copydoc MQTT_GetConnectPacketSize
+
+@page mqtt_serializeconnect_function MQTT_SerializeConnect
+@snippet mqtt_lightweight.h declare_mqtt_serializeconnect
+@copydoc MQTT_SerializeConnect
+
+@page mqtt_getsubscribepacketsize_function MQTT_GetSubscribePacketSize
+@snippet mqtt_lightweight.h declare_mqtt_getsubscribepacketsize
+@copydoc MQTT_GetSubscribePacketSize
+
+@page mqtt_serializesubscribe_function MQTT_SerializeSubscribe
+@snippet mqtt_lightweight.h declare_mqtt_serializesubscribe
+@copydoc MQTT_SerializeSubscribe
+
+@page mqtt_getunsubscribepacketsize_function MQTT_GetUnsubscribePacketSize
+@snippet mqtt_lightweight.h declare_mqtt_getunsubscribepacketsize
+@copydoc MQTT_GetUnsubscribePacketSize
+
+@page mqtt_serializeunsubscribe_function MQTT_SerializeUnsubscribe
+@snippet mqtt_lightweight.h declare_mqtt_serializeunsubscribe
+@copydoc MQTT_SerializeUnsubscribe
+
+@page mqtt_getpublishpacketsize_function MQTT_GetPublishPacketSize
+@snippet mqtt_lightweight.h declare_mqtt_getpublishpacketsize
+@copydoc MQTT_GetPublishPacketSize
+
+@page mqtt_serializepublish_function MQTT_SerializePublish
+@snippet mqtt_lightweight.h declare_mqtt_serializepublish
+@copydoc MQTT_SerializePublish
+
+@page mqtt_serializepublishheader_function MQTT_SerializePublishHeader
+@snippet mqtt_lightweight.h declare_mqtt_serializepublishheader
+@copydoc MQTT_SerializePublishHeader
+
+@page mqtt_serializeack_function MQTT_SerializeAck
+@snippet mqtt_lightweight.h declare_mqtt_serializeack
+@copydoc MQTT_SerializeAck
+
+@page mqtt_getdisconnectpacketsize_function MQTT_GetDisconnectPacketSize
+@snippet mqtt_lightweight.h declare_mqtt_getdisconnectpacketsize
+@copydoc MQTT_GetDisconnectPacketSize
+
+@page mqtt_serializedisconnect_function MQTT_SerializeDisconnect
+@snippet mqtt_lightweight.h declare_mqtt_serializedisconnect
+@copydoc MQTT_SerializeDisconnect
+
+@page mqtt_getpingreqpacketsize_function MQTT_GetPingreqPacketSize
+@snippet mqtt_lightweight.h declare_mqtt_getpingreqpacketsize
+@copydoc MQTT_GetPingreqPacketSize
+
+@page mqtt_serializepingreq_function MQTT_SerializePingreq
+@snippet mqtt_lightweight.h declare_mqtt_serializepingreq
+@copydoc MQTT_SerializePingreq
+
+@page mqtt_deserializepublish_function MQTT_DeserializePublish
+@snippet mqtt_lightweight.h declare_mqtt_deserializepublish
+@copydoc MQTT_DeserializePublish
+
+@page mqtt_deserializeack_function MQTT_DeserializeAck
+@snippet mqtt_lightweight.h declare_mqtt_deserializeack
+@copydoc MQTT_DeserializeAck
+
+@page mqtt_getincomingpackettypeandlength_function MQTT_GetIncomingPacketTypeAndLength
+@snippet mqtt_lightweight.h declare_mqtt_getincomingpackettypeandlength
+@copydoc MQTT_GetIncomingPacketTypeAndLength
+*/
+
+/**
+@defgroup mqtt_enum_types Enumerated Types
```

Comment:
For enums, I see an overview like:
```
enum  MQTTSubAckStatus_t { MQTTSubAckSuccessQos0 = 0x00, MQTTSubAckSuccessQos1 = 0x01, MQTTSubAckSuccessQos2
= 0x02, MQTTSubAckFailure = 0x80 }
```
and later a detailed description:

|Enumerator | |
|--|--|
|MQTTSubAckSuccessQos0 | Success with a maximum delivery at QoS 0 . |
|MQTTSubAckSuccessQos1 | Success with a maximum delivery at QoS 1. |
|MQTTSubAckSuccessQos2 | Success with a maximum delivery at QoS 2. |
|MQTTSubAckFailure | Failure. |

I wonder if we can have linebreaks in the overview or another column (e.g. "value (if defined)") in the description, since we might want to better convey that some enums like this one also have meaningful values.

Reply:
Yea it appears doxygen doesn't easily expand enums like macros since they have only a C preprocessor and not a compiler.